bedms/train.py

```diff
@@ -0,0 +1,242 @@
+""" This is the training script with which the user can train their own models."""
+
+import logging
+import torch
+from torch import nn
+from torch import optim
+from sklearn.metrics import (
+    precision_score,
+    recall_score,
+    f1_score,
+)
+import yaml
+from .utils_train import (
+    load_from_dir,
+    accumulate_data,
+    training_encoding,
+    data_loader,
+    train_model,
+    plot_learning_curve,
+    model_testing,
+    plot_confusion_matrix,
+    auc_roc_curve,
+)
+from .const import PROJECT_NAME
+from .model import BoWSTModel
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(PROJECT_NAME)
+
+
+class AttrStandardizerTrainer:
```

Comment:
I like that class, but the fact that no methods take any arguments makes me think they should just run automatically. The user has no control over anything so they now need to just run the functions... For example:

```python
trainer = AttrStandardizerTrainer("path/to/config.yaml")
trainer.load_encode_data()
trainer.train()
trainer.test()
```
I suppose the `train` and `test` **should not** run automatically of course, but I guess I'll argue that i think `load_encode_data` should just auto-run since they have to do it anyways... unless its a ton of data